compiler_gym/third_party/LLVM-Canon/README.md

```diff
@@ -0,0 +1,2 @@
+# LLVM-Canon
+LLVM-Canon aims to transform LLVM modules into a canonical form by reordering and renaming instructions while preserving the same semantics.
```

Comment:
Could you please extend this to follow [this template](https://github.com/facebookresearch/CompilerGym/tree/development/compiler_gym/third_party/autophase)? Specifically, add an author list to credit @michalpaszkowski, the URL of the upstream project https://github.com/michalpaszkowski/LLVM-Canon, and hash of the git commit this version is based on. Thanks!

Reply:
Done